rclpy/src/rclpy/wait_set.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY__WAIT_SET_HPP_
+#define RCLPY__WAIT_SET_HPP_
+
+#include <pybind11/pybind11.h>
+
+#include <string>
+
+namespace py = pybind11;
+
+namespace rclpy
+{
+/// Return a Capsule pointing to a zero initialized rcl_wait_set_t structure
+py::capsule
+get_zero_initialized_wait_set();
+
+/// Initialize a wait set
+/**
+ * Raises RCLError if the wait set could not be initialized
+ *
+ * \param[in] pywait_set Capsule pointing to the wait set structure
+ * \param[in] node_name string name of the node to be created
+ * \param[in] number_of_subscriptions int
+ * \param[in] number_of_guard_conditions int
+ * \param[in] number_of_timers int
+ * \param[in] number_of_clients int
+ * \param[in] number_of_services int
+ */
+void
+wait_set_init(
+  py::capsule pywait_set,
+  size_t number_of_subscriptions,
+  size_t number_of_guard_conditions,
+  size_t number_of_timers,
+  size_t number_of_clients,
+  size_t number_of_services,
+  size_t number_of_events,
+  py::capsule pycontext);
+
+/// Clear all the pointers in the wait set
+/**
+ * Raises RCLError if any rcl error occurs
+ *
+ * \param[in] pywait_set Capsule pointing to the wait set structure
+ */
+void
+wait_set_clear_entities(py::capsule pywait_set);
+
+/// Add an entity to the wait set structure
+/**
+ * Raises RuntimeError if the entity type is unknown
+ * Raises RCLError if any lower level error occurs
+ *
+ * \param[in] entity_type string defining the entity ["subscription, client, service"]
+ * \param[in] pywait_set Capsule pointing to the wait set structure
+ * \param[in] pyentity Capsule pointing to the entity to add
+ * \return Index in waitset entity was added at
+ */
+size_t
+wait_set_add_entity(const std::string entity_type, py::capsule pywait_set, py::capsule pyentity);
```

Comment:
Is there any value in the `const` qualifier for the string here? If it were passed by reference, I could see a performance gain, but I think it's just clutter in the header on a parameter that's passed by value, right?

Reply:
I added it to ask the compiler to double check that I wasn't assigning to entity_type instead of doing comparisons. I can remove it.